src/pubsub.ts

```diff
@@ -276,7 +276,12 @@ export class PubSub {
     this.determineBaseUrl_();
     this.api = {};
     this.auth = new GoogleAuth(this.options);
+
     this.projectId = this.options.projectId || PROJECT_ID_PLACEHOLDER;
+    if (!('projectId' in this.options) && process.env.PUBSUB_EMULATOR_HOST && process.env.GOOGLE_CLOUD_PROJECT) {
```

Comment:
Could you expand on this a little bit? Seems like this PR is hoping to explicitly prefer the `GOOGLE_CLOUD_PROJECT` env var only when using the emulator. I think this might end up giving unexpected behavior when not using the emulator. 

Why not just do something like

```js
this.projectId = this.options.projectId || process.env.GOOGLE_CLOUD_PROJECT || PROJECT_ID_PLACEHOLDER;
```

Reply:
It doesn't prefer it. It only uses `GOOGLE_CLOUD_PROJECT` for `projectId` if it hasn't been passed manually.

> Why not just do something like
>
> ```js
> this.projectId = this.options.projectId || process.env.GOOGLE_CLOUD_PROJECT || PROJECT_ID_PLACEHOLDER;
> ```

Because this would override manually passed *falsy values*, such as `''`, `0`, `null`, `undefined`. It didn't want to make the assumption of interpreting falsy values as "pick the env var or the default one".

Reply:
If the projectId is not set we attempt to pull it in from the auth library. Setting it in this way (when using the emulator) means that we are preferring it over whatever the auth library would otherwise provide.

Reply:
Ah, my mistake, looks like we don't pull the project id from the auth library when using the emulator. I think maybe a better fix would be to always pull the project ID from the auth library regardless of whether or not we're using the emulator. 

@stephenplusplus @bcoe thoughts?

Reply:
Hi there, I might be wrong but I assume that the variable should be `PUBSUB_PROJECT_ID`? Just like there is `DATASTORE_PROJECT_ID` when using the Datastore emulator?

Thanks!